src/main/java/edu/ucla/library/bucketeer/verticles/ClearCacheVerticle.java

```diff
@@ -0,0 +1,79 @@
+package edu.ucla.library.bucketeer.verticles;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import edu.ucla.library.bucketeer.Config;
+import edu.ucla.library.bucketeer.Constants;
+import edu.ucla.library.bucketeer.MessageCodes;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+
+
+
+/**
+ * A verticle to clear cantaloupe cache
+ */
+public class ClearCacheVerticle extends AbstractVerticle {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClearCacheVerticle.class, Constants.MESSAGES);
+
+    private String myUsername;
+
+    private String myPassword;
+
+    /**
+     * Clears cantaloupe cache and sends message if failure
+     */
+    @Override
+    public void start() throws Exception {
+        super.start();
+
+        final WebClient client = WebClient.create(Vertx.vertx());
+        final JsonObject config = config();
+
+        myUsername = config.getString(Config.IIIF_CACHE_USER);
+        myPassword = config.getString(Config.IIIF_CACHE_PASSWORD);
+
+        getJsonConsumer().handler(response -> {
+            final String imageID = response.body().getString("imageID");
+
+            // This will eventually be a feature flag but this solves the issue for now
+            if (myUsername == null && myPassword == null) {
+                response.reply(MessageCodes.BUCKETEER_603);
+            } else if (imageID == null) {
+                response.reply(MessageCodes.BUCKETEER_604);
+            } else {
+                client.postAbs("https://test.iiif.library.ucla.edu/tasks")
+                    .basicAuthentication(myUsername, myPassword)
+                    .putHeader("content-type", "application/json")
+                    .sendJsonObject(new JsonObject()
+                    .put("verb", "PurgeItemFromCache").put("identifier", imageID), post -> {
+                        if (post.succeeded()) {
+                            LOGGER.info(Integer.toString(post.result().statusCode()));
+                            response.reply(response.body());
+                        } else {
+                            LOGGER.error(post.cause(), post.cause().getMessage());
+                            response.fail(post.result().statusCode(), post.cause().getMessage());
+                        }
```

Comment:
I probably(?) also steered her here since I don't think that Cantaloupe returns anything other than the 202, regardless of whether the cache is actually cleared. I didn't test this, but am relying on memory. But I agree, on reflection, it would be better to test for bad responses here too. If not relevant now, Cantaloupe could return them in the future.

Reply:
Oh, I do remember... it wasn't that the linked docs page doesn't have any response codes other than 202, but that I've submitted bad IDs to Cantaloupe and it's happily returned 202s for those too.

Reply:
I see what you mean that Cantaloupe does, rather unconditionally, return a 202 here, but it will return a 401 if the cache credentials are missing or incorrect (I have not filled the placeholder values on my local yet the test still passes), or a 400 [if the response body does not represent a valid command](https://github.com/cantaloupe-project/cantaloupe/blob/0e186d74120606ba5b3678e91035f09a609984b0/src/main/java/edu/illinois/library/cantaloupe/resource/api/TasksResource.java#L50).

Reply:
The issue as you both pointed out is Cantaloupe's consistency in sending a 202 response. If the response isn't 202, should the response fail?

Reply:
Yes